lib/sift/client.rb

```diff
@@ -67,94 +69,147 @@ def original_request
-    #   Overrides the default API path with a different URL.
+    #   :api_key::
+    #     Overrides the API key for this call.
     #
-    # return_score (optional)
-    #   Whether the API response should include a score for this user. The score will
-    #   be calculated using the submitted event.  This feature must be
-    #   enabled for your account in order to use it.  Please contact
-    #   support@siftscience.com if you are interested in using this feature.
+    #   :version::
+    #     Overrides the version of the Events API to call.
     #
-    # return_action (optional)
-    #   Whether the API response should include an action triggered for this transaction.
+    #   :path::
+    #     Overrides the URI path for this API call.
     #
-    # == Returns:
-    #   In the case of an HTTP error (timeout, broken connection, etc.), this
-    #   method returns nil; otherwise, a Response object is returned and captures
-    #   the status message and status code. In general, you can ignore the returned
-    #   result, though.
+    # ==== Returns:
     #
-    def track(event, properties = {}, timeout = nil, path = nil, return_score = false, api_key = @api_key, return_action = false)
-      warn "[WARNING] api_key cannot be empty, fallback to default api_key." if api_key.to_s.empty?
-      api_key ||= @api_key
+    # In the case of an HTTP error (timeout, broken connection, etc.),
+    # this method returns nil; otherwise, a Response object is
+    # returned and captures the status message and status code. In
+    # general, you can ignore the returned result, though.
+    #
+    def track(event, properties = {}, opts = {})
+      api_key = opts[:api_key] || @api_key
+      version = opts[:version] || @version
+      path = opts[:path] || (version && Sift.rest_api_path(version)) || @path
+      timeout = opts[:timeout] || @timeout
+      return_score = opts[:return_score]
+      return_action = opts[:return_action]
+      return_workflow_status = opts[:return_workflow_status]
+      abuse_types = opts[:abuse_types]
+
       raise("event must be a non-empty string") if (!event.is_a? String) || event.empty?
       raise("properties cannot be empty") if properties.empty?
-      raise("Bad api_key parameter") if api_key.empty?
-      path ||= @path
-      timeout ||= @timeout
+      raise("api_key cannot be empty") if api_key.empty?
 
-      uri = URI.parse(API_ENDPOINT)      
-      uri.query = URI.encode_www_form(URI.decode_www_form(uri.query.to_s) << ["return_score", "true"]) if return_score
-      uri.query = URI.encode_www_form(URI.decode_www_form(uri.query.to_s) << ["return_action", "true"]) if return_action
-      path = path + "?" + uri.query if !uri.query.to_s.empty?
+      query = {}
+      query["return_score"] = "true" if return_score
+      query["return_action"] = "true" if return_action
+      query["return_workflow_status"] = "true" if return_workflow_status
+      query["abuse_types"] = abuse_types.join(",") if abuse_types
```

Comment:
Do we want to enforce in the client that the customer can only append one of {return_score, return_action, return_worflow_status} ?